flickit-assessment-server/src/test/java/org/flickit/assessment/scenario/test/users/space/CreateSpaceScenarioTest.java

```diff
@@ -67,4 +67,38 @@ void createSpace_duplicateTitle() {
         int countAfter = jpaTemplate.count(SpaceJpaEntity.class);
         assertEquals(countBefore, countAfter);
     }
+
+    @Test
+    void createSpace_withSameTitleAsDeleted() {
+        final var request = createSpaceRequestDto();
+        var firstCreateResponse = spaceHelper.create(context, request);
+        firstCreateResponse.then()
+            .statusCode(201);
+
+        final int countBeforeDelete = jpaTemplate.count(SpaceJpaEntity.class);
+        var createdSpaceId = firstCreateResponse.body().path("id");
+        SpaceJpaEntity firstSpace = jpaTemplate.load(createdSpaceId, SpaceJpaEntity.class);
+
+        var deleteResponse = spaceHelper.delete(context, createdSpaceId.toString());
+        deleteResponse.then()
+            .statusCode(204);
+
+        final int countAfterDelete = jpaTemplate.count(SpaceJpaEntity.class);
+        SpaceJpaEntity deletedSpace = jpaTemplate.load(createdSpaceId, SpaceJpaEntity.class);
+
+        var secondCreateResponseWithSameTitle = spaceHelper.create(context, request);
+        secondCreateResponseWithSameTitle.then()
+            .statusCode(201);
+
+        final int countAfterSecondCreation = jpaTemplate.count(SpaceJpaEntity.class);
+        var newSpaceId = secondCreateResponseWithSameTitle.body().path("id");
+        SpaceJpaEntity secondSpace = jpaTemplate.load(newSpaceId, SpaceJpaEntity.class);
+
+        assertFalse(firstSpace.isDeleted());
```

Comment:
We’re not testing the delete space scenario here, so these assertions aren’t necessary.